### PR TITLE
Added .NET Core supported environment variables

### PR DESCRIPTION
**Pull request by ahardin-rh**
Continues work from https://github.com/openshift/openshift-docs/pull/3955

Preview build:
http://file.rdu.redhat.com/~ahardin/03172017/jerboaa-fork/using_images/s2i_images/dot_net_core.html

### CODE REVIEW

**Comment by ahardin-rh** [comment]
@jerboaa @omajid PTAL

@adellape please peer review

**Comment by omajid** [comment]
👍

**Comment by jerboaa** [comment]
@ahardin-rh This looks good. Thanks!

File: using_images/s2i_images/dot_net_core.adoc [hunk 1]
```diff
@@ -66,8 +66,73 @@ link:https://github.com/redhat-developer/s2i-dotnetcore/blob/master/dotnet_image
...
+
+.NET Core Environment Variables
+[cols="4a,6a,6a",options="header"]
+|===
+
+|Variable Name |Description |Default
+
+|`DOTNET_STARTUP_PROJECT`
+|Used to select the project to run. This must be the folder containing *_project.json_*.
+|Current working directory.
```
Comment (tmds) [inline using_images/s2i_images/dot_net_core.adoc]: @jerboaa @ahardin-rh 
'Current working directory' may be confusing for end-users. This is from the perspective of the builder.
Suggestion:
Default: `.`
Description: This must be the folder  in the source repository containing *_project.json_*.

Comment (jerboaa) [inline using_images/s2i_images/dot_net_core.adoc]: @tmds To be honest, I don't see any difference between "Current working directory" and '.' How is '.' clearer? It's the same with the latter slightly more confusing to me. I'm fine with adding some blurb which tells the user that this is from perspective of the "builder". Yet, all of those env vars are basically build env config.

Comment (tmds) [inline using_images/s2i_images/dot_net_core.adoc]: It's made more clear by adding "in the source repository" in the description.

Current directory may mean something else to the user:
```
~/repos/new-webapp/src (master)$ oc new-app ...
```

**Comment by ahardin-rh** [comment]
@tmds @jerboaa This is updated.

**Comment by ahardin-rh** [comment]
@bmcelvee @mburke5678 please peer review?

File: using_images/s2i_images/dot_net_core.adoc [hunk 2]
```diff
@@ -66,8 +66,73 @@ link:https://github.com/redhat-developer/s2i-dotnetcore/blob/master/dotnet_image
...
 
-The .NET Core image currently does not support any environment variables. No
-configuration is necessary.
+The .NET Core images support a number of environment variables, which you can
+set to control the build behavior of your .NET Core application.
+
+[NOTE]
+====
+You must set environment variables that control build behavior in the s2i build
+configuration or in the *_.s2i/environment_* file in to make them available to
```
Comment (mburke5678) [inline using_images/s2i_images/dot_net_core.adoc]: s/in to make/to make
-or-
s/in to make/in order to make

**Comment by mburke5678** [comment]
@ahardin-rh One small fix.

File: using_images/s2i_images/dot_net_core.adoc [hunk 3]
```diff
@@ -66,8 +66,73 @@ link:https://github.com/redhat-developer/s2i-dotnetcore/blob/master/dotnet_image
...
 == Configuration
 
-The .NET Core image currently does not support any environment variables. No
-configuration is necessary.
+The .NET Core images support a number of environment variables, which you can
+set to control the build behavior of your .NET Core application.
+
+[NOTE]
+====
+You must set environment variables that control build behavior in the s2i build
```
Comment (adellape) [inline using_images/s2i_images/dot_net_core.adoc]: s/s2i/S2I/

File: using_images/s2i_images/dot_net_core.adoc [hunk 4]
```diff
@@ -66,8 +66,73 @@ link:https://github.com/redhat-developer/s2i-dotnetcore/blob/master/dotnet_image
...
+
+.NET Core Environment Variables
+[cols="4a,6a,6a",options="header"]
+|===
+
+|Variable Name |Description |Default
+
+|`DOTNET_STARTUP_PROJECT`
+|Used to select the project to run. This must be the folder in the source repository containing *_project.json_*.
+|.
```
Comment (adellape) [inline using_images/s2i_images/dot_net_core.adoc]: If this is an intentional `.` meaning the current directory, maybe use filename/directory markup.

File: using_images/s2i_images/dot_net_core.adoc [hunk 5]
```diff
@@ -66,8 +66,73 @@ link:https://github.com/redhat-developer/s2i-dotnetcore/blob/master/dotnet_image
...
+
+|`DOTNET_PUBLISH`
+|Used to control whether the application should be built by executing `dotnet
+build` or `dotnet publish`. To publish the application, set the value to `true`.
+It is recommended to publish your application. |For backwards compatibility, the
```
Comment (adellape) [inline using_images/s2i_images/dot_net_core.adoc]: `|For`... can start on a new line.

**Comment by adellape** [comment]
@ahardin-rh Couple comments inline, and a just-in-case reminder: remember to drop `[Do not merge]` from title when you're ready before merging.

File: using_images/s2i_images/dot_net_core.adoc (hunk 5 above)
Comment (ahardin-rh) [inline using_images/s2i_images/dot_net_core.adoc]: Woops!

**Comment by jerboaa** [comment]
Looks OK to me.

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../using_images/s2i_images/dot_net_core.adoc#using-images-using-dot-net-core[Source-to-Image (S2I) -> .NET Core
|Added .NET Core supported xref:../using_images/s2i_images/dot_net_core.adoc#dot-net-core-configuration[environment variables].
%

**Pull request merged**